pydantic/main.py

```diff
@@ -1473,6 +1474,19 @@ def create_model(  # noqa: C901
                     'Field definitions should be a `(<type>, <default>)`.',
                     code='create-model-field-definitions',
                 ) from e
+
+        elif _typing_extra.is_annotated(f_def):
+            (f_annotation, f_value, *_) = typing_extensions.get_args(
+                f_def
+            )  # first two input are expected from Annotated
+            from .fields import FieldInfo
+
+            if not isinstance(f_value, FieldInfo):
+                raise PydanticUserError(
+                    'Field definitions should be a Annotated[<type>, FieldInfo(...)]',
+                    code='create-model-field-definitions',
```

Comment:
Could you please add an example to the `usage errors` section of the docs showing an example of how using `Annotated` with a field def incorrectly leads to this error being raised? See https://docs.pydantic.dev/latest/errors/usage_errors/#create-model-field-definitions

Reply:
Sounds good, will add the error documentation!